eng/common/pipelines/templates/archetype-typespec-emitter.yml

```diff
@@ -189,26 +193,25 @@ extends:
                 ServiceEndpointUrl: https://api.esrp.microsoft.com
                 MainPublisher: ESRPRELPACMANTEST
 
-          - task: PowerShell@2
-            displayName: Create emitter-package.json
-            inputs:
-              pwsh: true
-              filePath: ./eng/common/scripts/typespec/New-EmitterPackageJson.ps1
-              arguments: >
-                -PackageJsonPath '$(buildArtifactsPath)/lock-files/package.json'
-                -OverridesPath '$(buildArtifactsPath)/overrides.json'
-                -OutputDirectory '$(Build.ArtifactStagingDirectory)'
-              workingDirectory: $(Build.SourcesDirectory)
+          - pwsh: |
+              npm install -g @azure-tools/typespec-client-generator-cli@latest
+            displayName: Install tsp-client
 
-          - task: PowerShell@2
-            displayName: Create emitter-package-lock.json
-            inputs:
-              pwsh: true
-              filePath: ./eng/common/scripts/typespec/New-EmitterPackageLock.ps1
-              arguments: >
-                -EmitterPackageJsonPath '$(Build.ArtifactStagingDirectory)/emitter-package.json'
-                -OutputDirectory '$(Build.ArtifactStagingDirectory)'
-              workingDirectory: $(Build.SourcesDirectory)
+          - pwsh: |
+              if (Test-Path -Path '$(buildArtifactsPath)/overrides.json') {
```

Comment:
What is the scenario where we need to override the emitter-package.json? 

cc @m-nash

Reply:
I also chatted with @m-nash about this and I can understand using this for cases where we want to transition or use a completely different emitter but I want to make sure it is only temporary so that we don't get into a place where service directories are all using different versions.

Reply:
These aren't going to be defined at the service level. There will be 1 emitter-package.json for the old emitter and another one for the new emitter. We do eventually want to support service level customization of dependencies, but this would likely use tsp-location.yaml.